x-pack/plugins/uptime/server/lib/adapters/telemetry/kibana_telemetry_adapter.ts

```diff
@@ -28,22 +28,153 @@ export class KibanaTelemetryAdapter {
+    } else {
+      const prevDateEnd = [...bucket.dateRangeEnd].pop();
+      if (!prevDateEnd || prevDateEnd !== dateEnd) {
+        bucket.dateRangeStart.push(dateStart);
+        bucket.dateRangeEnd.push(dateEnd);
+      }
+    }
+
+    const prevAutorefreshInterval = [...bucket.autorefreshInterval].pop();
+    if (!prevAutorefreshInterval || prevAutorefreshInterval !== autorefreshInterval) {
+      bucket.autorefreshInterval.push(autorefreshInterval);
+    }
+    bucket.autoRefreshEnabled = autoRefreshEnabled;
   }
 
-  public static countMonitor() {
+  public static async countNoOfUniqueMonitorAndLocations(callCluster: APICaller) {
+    const params = {
+      index: INDEX_NAMES.HEARTBEAT,
+      body: {
+        query: {
+          bool: {
+            must: [
+              {
+                range: {
+                  '@timestamp': {
+                    gte: 'now-1d/d',
+                    lt: 'now',
+                  },
+                },
+              },
+            ],
+          },
+        },
+        size: 0,
+        aggs: {
+          unique_monitors: {
+            cardinality: {
+              field: 'monitor.id',
+            },
+          },
+          unique_locations: {
+            cardinality: {
+              field: 'observer.geo.name',
+              missing: 'N/A',
+            },
+          },
+          monitor_name: {
+            string_stats: {
+              field: 'monitor.name',
+            },
+          },
+          observer_loc_name: {
+            string_stats: {
+              field: 'observer.geo.name',
+            },
+          },
+          monitors: {
+            terms: {
+              field: 'monitor.id',
+              size: 10000,
+            },
+            aggs: {
+              docs: {
+                top_hits: {
+                  size: 1,
+                  _source: ['monitor.timespan'],
+                },
+              },
+            },
+          },
+        },
+      },
+    };
+
+    const result = await callCluster('search', params);
+    const numberOfUniqueMonitors: number = result?.aggregations?.unique_monitors?.value ?? 0;
+    const numberOfUniqueLocations: number = result?.aggregations?.unique_locations?.value ?? 0;
+    const monitorName: any = result?.aggregations?.monitor_name;
+    const locationName: any = result?.aggregations?.observer_loc_name;
+    const uniqueMonitors: any = result?.aggregations?.monitors.buckets;
     const bucket = this.getBucketToIncrement();
-    this.collector[bucket].monitor_page += 1;
+
+    this.collector[bucket].no_of_unique_monitors = numberOfUniqueMonitors;
+    this.collector[bucket].no_of_unique_observer_locations = numberOfUniqueLocations;
+    this.collector[bucket].no_of_unique_observer_locations = numberOfUniqueLocations;
+    this.collector[bucket].monitor_name_stats = {
+      min_length: monitorName.min_length,
+      max_length: monitorName.max_length,
+      avg_length: +monitorName.avg_length.toFixed(2),
```

Comment:
This seems like a potential `TypeError` to me, but I haven't tested it.

Reply:
i have tested it, it works

Reply:
@shahzad31 sorry, I thought I had deleted this comment. I tested it too, and meant to remove the comment from the pending review.